Makefile

```diff
@@ -185,6 +185,35 @@ DEFINES += \
     WEAVE_DEVICE_CONFIG_DEVICE_FIRMWARE_REVISION=\"$(DEVICE_FIRMWARE_REVISION)\"
 endif
 
+# A product's firmware usually comes with at least two build "flavors":
+# 1) a "development" build which typically enables debugging artifacts along with logging, and disables
+# security and optimizations and 2) a "release" build which targets end-user devices
+# and disables debugging artifacts and logging, and enables security and optimizations.
+#
+# To facilitate supporting these two build flavors, the sample app source code supports
+# the Makefile variable BUILD_DEVELOPMENT. BUILD_DEVELOPMENT is enabled by default so that
+# with no command line argument pecified at compile-time, a development build is produced.
+# Specifying BUILD_DEVELOPMENT=0 when running make which will yield a "pseudo" release build.
+#
+#
+# IMPORTANT NOTE: we say "pseudo" release build because the sample app is never to be used
```

Comment:
Maybe reword slightly:

... is never to be used **as-is** as the firmware for an end-user device.

Reply:
done